Makefile

```diff
@@ -1,8 +1,8 @@
 CC = gcc
 CXX = g++
 
-CFLAGS = -W -Wall -Wextra -ansi -pedantic -lm -O2
-CXXFLAGS = -W -Wall -Wextra -ansi -pedantic -O2
+override CFLAGS := -W -Wall -Wextra -ansi -pedantic -lm -O2 $(CFLAGS)
```

Comment:
Because the idea is to append the user specified ones to flags in `Makefile`, not the other way around. Using `+=` would require renaming `CFLAGS` in `Makefile` to something else which seems unnecessary churn compared to the approach in this commit to me.